dev/Decimal/decimalcppwinrt.h

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) Microsoft Corporation and Contributors.
+// Licensed under the MIT License.
+
+#include <oleauto.h>
+
+#if defined(WINRT_Microsoft_Windows_Foundation_H) && !defined(__WINDOWSAPPSDK_WIMRT_M_W_F_DECIMAL_)
+#define __WINDOWSAPPSDK_WIMRT_M_W_F_DECIMAL_
+
+//----------------------------------------------------------------------
+// WinRT Decimal operators
+
+inline bool operator==(
+    winrt::Microsoft::Windows::Foundation::Decimal const& left,
+    winrt::Microsoft::Windows::Foundation::Decimal const& right)
+{
+    return left.Compare(right) == 0;
+}
+
+inline bool operator!=(
+    winrt::Microsoft::Windows::Foundation::Decimal const& left,
+    winrt::Microsoft::Windows::Foundation::Decimal const& right)
+{
+    return left.Compare(right) != 0;
+}
+
+inline bool operator<(
+    winrt::Microsoft::Windows::Foundation::Decimal const& left,
+    winrt::Microsoft::Windows::Foundation::Decimal const& right)
+{
+    return left.Compare(right) < 0;
+}
+
+inline bool operator<=(
+    winrt::Microsoft::Windows::Foundation::Decimal const& left,
+    winrt::Microsoft::Windows::Foundation::Decimal const& right)
+{
+    return left.Compare(right) <= 0;
+}
+
+inline bool operator>(
+    winrt::Microsoft::Windows::Foundation::Decimal const& left,
+    winrt::Microsoft::Windows::Foundation::Decimal const& right)
+{
+    return left.Compare(right) > 0;
+}
+
+inline bool operator>=(
+    winrt::Microsoft::Windows::Foundation::Decimal const& left,
+    winrt::Microsoft::Windows::Foundation::Decimal const& right)
+{
+    return left.Compare(right) >= 0;
+}
+
+//----------------------------------------------------------------------
+// Conversion functions
+
+namespace winrt::Microsoft::Windows::Foundation
+{
+/// Return value as DECIMAL.
+inline DECIMAL to_DECIMAL(winrt::Microsoft::Windows::Foundation::DecimalValue const& value)
+{
+    return *reinterpret_cast<DECIMAL const*>(&value);
```

Comment:
Recommend avoiding UB here and using either std::bit_cast or memcpy.

Reply:
UB = ?

Reply:
Undefined behavior